src/components/password-modal/password-modal.tsx

```diff
@@ -0,0 +1,135 @@
+                        onPress={() => {
+                            this.onReject && this.onReject();
+                        }}
+                    />
+                </View>
+                <View style={this.props.styles.container}>
+                    <View style={this.props.styles.topContainer}>
+                        <Text large style={{ textAlign: 'center' }}>
+                            {this.props.infoText}
+                        </Text>
+                        <View style={this.props.styles.inputBox}>
+                            <TextInput
+                                testID="input-password"
+                                style={this.props.styles.input}
+                                placeholderTextColor={this.props.theme.colors.textSecondary}
+                                placeholder={translate('SetPassword.password')}
+                                autoCapitalize={'none'}
+                                autoCorrect={false}
+                                selectionColor={this.props.theme.colors.accent}
+                                secureTextEntry={!this.state.revealPassword}
+                                value={this.state.password}
+                                onChangeText={password => {
+                                    this.setState({
+                                        password
+                                    });
+                                }}
+                            />
+                            <TouchableHighlight
+                                testID="reveal-password"
+                                onPress={() => {
+                                    this.setState({
+                                        revealPassword: !this.state.revealPassword
+                                    });
+                                }}
+                            >
+                                <Icon
+                                    name={this.state.revealPassword ? 'view-1' : 'view-off'}
+                                    size={18}
+                                    style={this.props.styles.icon}
+                                />
+                            </TouchableHighlight>
+                        </View>
+                    </View>
+                    <View style={this.props.styles.bottomContainer}>
+                        <Button
+                            style={this.props.styles.bottomButton}
+                            primary
+                            onPress={() => {
+                                this.onConfirm && this.onConfirm(this.state.password);
+                            }}
+                        >
+                            {this.props.buttonLabel}
+                        </Button>
+                    </View>
+                </View>
+            </Modal>
+        );
+    }
+}
+
+export const PasswordModal = withTheme(stylesProvider)(PasswordModalComponent);
```

Comment:
Even if you are using only one enhancer, you could use smartConnect, and add PropTypes, this way we get autocomplete functionality for component props. (See docs)